Krun Tuning

fsfod opened the pull request:
Try to get one iteration of each benchmark to run for 0.1 seconds.

vext01 commented:
So that's 0.1 seconds for which variant of the VM?

Is that on the machine you intend to run the preliminary results on?

Can you show me a Krun results file from running 1 process execution with say 10 in-process executions?

Any idea why git thinks the whole config file has changed?

fsfod commented:
The line endings got flipped when fixing a merge conflict I would of normally spotted that normally.

vext01 commented:
OK, well if possible, can we push a fix, so I can see what really changed.

Thanks

fsfod commented:
i would have to force push if thats what you really want

vext01 commented:
That's fine in this one instance.

vext01 commented:
That's better. Now, how many in-process iterations did you run to calibrate. The first few iterations are bound to be super slow. You'd want to let it run a little before looking at the times.

fsfod commented:
I don't know what machine its going to run on i just had guess based on my linux machine time that is clocked higher than most servers but is a few generations old(sandy bridge 2600k).

fsfod commented:
I don't know what machine its going to run on i just had guess based on my linux machine time that is clocked higher than most servers but is a few generations old(sandy bridge 2600k).

vext01 commented on simplerunner.lua:
```diff
@@ -57,7 +57,7 @@ function run_single_benchmarks(benchname)
 
 if arg[1] then
-    run_single_benchmarks(arg[1])
+    run_single_benchmarks(arg[1], arg[2])
```
Is this recursive?

vext01 replied:
ah i see, ignore me.

fsfod replied:
what do you mean that's the optional run count it can be nil and default to 1 inside runbench

vext01 replied:
ignore this comment. I misread the code.

vext01 commented:
Do you want to squash?

fsfod commented:
> Do you want to squash?
Do you have any comments on the JIT vs no JIT iteration times or anything?

vext01 commented:
> Do you have any comments on the JIT vs no JIT iteration times or anything?

Well, I saw you had reduced the count. Presumably you experimented to find a suitable value?

fsfod commented:
it was just some guess, but here are the slow downs for no JIT. The extra time for one iteration and slow down amount
```
binarytrees        0.077231, 1.587435
capnproto_decode   0.333185, 3.374063
capnproto_encode   1.635851, 13.863944
fannkuch_redux     0.331934, 3.672644
fasta              0.213632, 2.698880
jsonlua_decode     0.184182, 2.335364
jsonlua_encode     0.013045, 1.099005
luacheck           -0.512662, 0.496973
luacheck_parser    0.026105, 1.183490
luafun             0.145883, 1.862993
md5                2.219446, 20.224372
nbody              0.985365, 7.003201
richards           0.348244, 3.225598
series             0.079451, 1.231476
spectralnorm       2.088578, 15.354350
```

vext01 commented:
What are the two columns sorry? One figure is negative. Is that right?

vext01 commented:
The second column "slow down amount", is how many times the jitted version?

vext01 commented:
I think you should estimate now, as I did in mattermost, how long the experiment will take to run based on your above estimates.

fsfod commented:
> What are the two columns sorry? One figure is negative. Is that right?

Yes lua check runs faster with the jit off. The first is extra seconds a single no JIT iteration takes over the Normal JIT. The other is how many times slower no JIT vs JIT

using n_pexecs, n_iters = 10, 1500 but n_iters for NoJIT = 100. 
Times are based on the average time of one iteration of all the benchmarks added together for each VM. 
Time in hours.
```
Normal             13.224481
DualNum            13.562758
NoJIT              3.150929
GC64               13.835596
Total              43.773763
```

ltratt commented:
OK, so I'd like to know why there's a negative time here too:

```
luacheck           -0.512662, 0.496973
```

fsfod commented:
>yes luacheck runs faster with the jit off

ltratt commented:
Aha, I see.

vext01 commented:
Well, your time estimate sounds good (In reality, krun will take a little longer to run these, as it has a few waits as part of normal operation).

If Laurie has no other comments, are we ready to squash?

fsfod commented:
@vext01  I had squashed most of the commits leaving what I think import changes left in the history. I also added back the pexce and iteration counts since you two are ok with the times

ltratt commented:
OK, I'll leave this to Edd to consider for merging.

vext01 commented:
OK. Ready to merge?

fsfod commented:
@vext01 yes merge it.
